File: peer-fixtures/textual/sb10_proof_app/sb10_proof_app_benchmark.py
from __future__ import annotations

import argparse
import asyncio

from textual.app import App, ComposeResult
from textual.widgets import Static


SCREENS = ["home", "search", "task", "logs", "diagnostics"]


def main() -> None:
    options = parse_args()
    if options.wire:
        WireProofApp(options).run()
    else:
        print("Textual SB.10 proof app fixture: use --wire for PTY capture")


class WireProofApp(App[None]):
    def __init__(self, options: argparse.Namespace) -> None:
        super().__init__()
        self.options = options
        self.step = 0
        self.active_screen = "home"
        self.events = ["boot proof app"]

    def compose(self) -> ComposeResult:
        yield Static(self.render_body(), id="body")

    def on_mount(self) -> None:
        self.body = self.query_one("#body", Static)
        asyncio.create_task(self.drive())

    async def drive(self) -> None:
        await asyncio.sleep(self.options.interval_ms / 1000)
        while self.step < self.options.steps:
            self.active_screen = SCREENS[self.step % len(SCREENS)]
            self.events.append(
                f"step={self.step} screen={self.active_screen} rows={self.options.rows}"
            )
            self.events = self.events[-12:]
            self.step += 1
            self.body.update(self.render_body())
            await asyncio.sleep(self.options.interval_ms / 1000)
        self.exit()

    def render_body(self) -> str:
        lines = [
            f"SB.10 proof app screen={self.active_screen} step={self.step}",
            "",
            f"nav: home search task  command: {command_name(self.active_screen):<14} status: {status(self.step)}",
            "",
            f"results visible={(self.step * 17) % self.options.rows} selected={self.step % 9}",
            "",
        ]
        lines.extend(self.events)
        return "\n".join(lines)


def command_name(screen: str) -> str:
    return {
        "home": "open-palette",
        "search": "rank-results",
        "task": "run-process",
        "logs": "copy-log",
    }.get(screen, "diagnose")


def status(step: int) -> str:
    return ["idle", "running", "complete", "warning"][step % 4]


def parse_args() -> argparse.Namespace:
    parser = argparse.ArgumentParser()
    parser.add_argument("--wire", action="store_true")
    parser.add_argument("--rows", type=positive_int, default=1000)
    parser.add_argument("--steps", type=positive_int, default=10)
    parser.add_argument("--interval-ms", type=positive_int, default=50)
    parser.add_argument("--size", default="120x32")
    return parser.parse_args()


def positive_int(value: str) -> int:
    parsed = int(value)
    if parsed <= 0:
        raise argparse.ArgumentTypeError("expected a positive integer")
    return parsed


if __name__ == "__main__":
    main()

File: peer-fixtures/ink/sb10_proof_app/bin/sb10_proof_app_benchmark.js
#!/usr/bin/env node

import React from 'react'
import { Box, Text, render as renderInk } from 'ink'

const screens = ['home', 'search', 'task', 'logs', 'diagnostics']
const options = parseArgs(process.argv.slice(2))
if (options.wire) await runWire(options)
else console.log('Ink SB.10 proof app fixture: use --wire for PTY capture')

async function runWire(options) {
  let instance
  const done = new Promise((resolve) => {
    instance = renderInk(
      React.createElement(WireProofApp, {
        options,
        onDone: () => {
          instance.unmount()
          resolve()
        },
      }),
      {
        exitOnCtrlC: false,
        interactive: true,
        alternateScreen: true,
        maxFps: 60,
        patchConsole: false,
      },
    )
  })
  await done
  await instance.waitUntilExit()
}

function WireProofApp({ options, onDone }) {
  const [step, setStep] = React.useState(0)
  const [events, setEvents] = React.useState(['boot proof app'])
  const screen = screens[step % screens.length]

  React.useEffect(() => {
    const timer = setInterval(() => {
      setStep((current) => {
        const next = Math.min(current + 1, options.steps)
        const nextScreen = screens[current % screens.length]
        setEvents((items) => [...items, `step=${current} screen=${nextScreen} rows=${options.rows}`].slice(-12))
        if (next >= options.steps) {
          clearInterval(timer)
          setTimeout(onDone, options.intervalMs)
        }
        return next
      })
    }, options.intervalMs)
    return () => clearInterval(timer)
  }, [options.steps, options.intervalMs, options.rows, onDone])

  return React.createElement(
    Box,
    { flexDirection: 'column' },
    React.createElement(Text, null, `SB.10 proof app screen=${screen} step=${step}`),
    React.createElement(Text, null, ''),
    React.createElement(
      Text,
      null,
      `nav: home search task  command: ${commandName(screen).padEnd(14)} status: ${status(step)}`,
    ),
    React.createElement(Text, null, ''),
    React.createElement(Text, null, `results visible=${(step * 17) % options.rows} selected=${step % 9}`),
    React.createElement(Text, null, ''),
    ...events.map((event, index) => React.createElement(Text, { key: index }, event)),
  )
}

function commandName(screen) {
  return {
    home: 'open-palette',
    search: 'rank-results',
    task: 'run-process',
    logs: 'copy-log',
  }[screen] ?? 'diagnose'
}

function status(step) {
  return ['idle', 'running', 'complete', 'warning'][step % 4]
}

function parseArgs(args) {
  const options = { wire: false, rows: 1000, steps: 10, intervalMs: 50 }
  for (const arg of args) {
    if (arg === '--wire') options.wire = true
    else if (arg.startsWith('--rows=')) options.rows = positive(arg, '--rows=')
    else if (arg.startsWith('--steps=')) options.steps = positive(arg, '--steps=')
    else if (arg.startsWith('--interval-ms=')) options.intervalMs = positive(arg, '--interval-ms=')
    else if (arg.startsWith('--size=')) continue
    else throw new Error(`unknown argument: ${arg}`)
  }
  return options
}

function positive(arg, prefix) {
  const value = Number(arg.slice(prefix.length))
  if (!Number.isInteger(value) || value <= 0) throw new Error(`${prefix} expects a positive integer`)
  return value
}
